contracts/contracts/strategies/NativeStaking/CompoundingValidatorManager.sol

```diff
@@ -514,26 +539,78 @@ abstract contract CompoundingValidatorManager is Governable {
         );
         // The verification slot must be after the deposit's slot.
         // This is needed for when the deposit queue is empty.
-        require(deposit.slot < verificationSlot, "Slot not after deposit");
-
-        // Calculate the next block timestamp from the verification slot.
-        uint64 nextBlockTimestamp = 12 *
-            verificationSlot +
-            BEACON_GENESIS_TIMESTAMP +
-            // Add 12 seconds for the next block
-            12;
-        // Get the parent beacon block root of the next block which is the block root of the verification slot.
+        require(
+            deposit.slot < depositVerificationSlot,
+            "Slot not after deposit"
+        );
+        require(
+            depositVerificationSlot <= validatorVerificationSlot,
+            "Invalid verification slots"
+        );
+
+        // Get the parent beacon block root of the next block which is the block root of the deposit verification slot.
         // This will revert if the slot after the verification slot was missed.
-        bytes32 blockRoot = BeaconRoots.parentBlockRoot(nextBlockTimestamp);
+        bytes32 depositBlockRoot = BeaconRoots.parentBlockRoot(
+            _calcNextBlockTimestamp(depositVerificationSlot)
+        );
 
         // Verify the slot of the first pending deposit matches the beacon chain
         bool isDepositQueueEmpty = IBeaconProofs(BEACON_PROOFS)
-            .verifyFirstPendingDepositSlot(
-                blockRoot,
-                firstPendingDepositSlot,
-                firstPendingDepositSlotProof
+            .verifyFirstPendingDeposit(
+                depositBlockRoot,
+                firstPendingDeposit.slot,
+                firstPendingDeposit.pubKeyHash,
+                firstPendingDeposit.pubKeyProof
+            );
+
```

Comment:
nit / style: we could do an early require:
```
require(!isDepositQueueEmpty, "Deposit queue empty");
```

removing an if statement below reducing number of code paths